alkemy/src/test/kotlin/io/alkemy/AssertionsWebElementTest.kt

```diff
@@ -7,7 +7,8 @@ import io.kotest.assertions.throwables.shouldThrow
 import io.kotest.core.spec.style.StringSpec
 import org.openqa.selenium.TimeoutException
 
-class AssertionsWebElementTest(val context: AlkemyContext) : StringSpec({
+class AssertionsWebElementTest : StringSpec({
+    val context = installAlkemyExtension()
```

Comment:
Just to let you know I'd have to reflect on this change before I can decide if it's the right thing for this framework. I valued the simplicity of autowiring the context, which is a common practice in test frameworks like [Micronaut test](https://micronaut-projects.github.io/micronaut-test/latest/guide/#kotest5), and it feels more elegant than manually adding this method call in each test. I'm not sure the benefits outweigh the drawbacks yet so I'll look at this closer in the following weeks.

Reply:
Sure, I understand, especially for such a major change.

The main advantage comes when customizing the configuration, see:
- https://github.com/cosmin-marginean/alkemy/pull/11/files#diff-5b65bb029624db4b333723cc602af40276185ba28bb368798339ed848739808aR8-R19
- https://github.com/cosmin-marginean/alkemy/pull/11/files#diff-2f48123ca0dcfc87f7fc9cf1b19a17d7b9ecaaec9ea363895d93454e087d9531L11-R29

Currently, customizing the configuration requires manually creating the `AlkemyContext`. But that doesn't register the `ReportContext` extension, and doesn't close the web drivers after the spec/project completes.

This approach uses the same way for creating the `AlkemyContext` whether or not the configuration is customized, which includes registering the `ReportContext` extension, and closing web drivers after the spec completes.

Reply:
Speaking of Micronaut, currently Alkemy is also not compatible with [io.micronaut.test:micronaut-test-kotest5](https://micronaut-projects.github.io/micronaut-test/snapshot/guide/index.html#kotest5) because it uses constructor injection (yes you can manually create an `AlkemyContext`, but then you still need manually register the `ReportContext` and manually close web drivers after the spec/project completes).

By making Alkemy a `MountableExtension`, Alkemy will work out of the box with any other Kotest extensions that use constructor injection. With this change, the only thing that the `alkemy-spring` module is doing is providing an easy way to automatically map from a Spring configuration properties to an `AlkemyConfig` instance. The same can be easily done for Micronaut, and is a lot simpler than registering `AlkemyContext` as a bean so that it can be injected in the constructor with other dependencies (see https://github.com/cosmin-marginean/alkemy/pull/11/files#diff-f445eaf4562146275b18e9c9042076afdc746958046a619c5556c76c03ea5725L31-R45).